tests/std/tests/P2231R1_complete_constexpr_optional_variant/test.cpp

```diff
@@ -0,0 +1,334 @@
+
+        optional<T> nullopt_constructed{nullopt};
+        assert(!nullopt_constructed.has_value());
+    }
+
+    { // construction from underlying type
+        const T input{42};
+        optional<T> copy_constructed{input};
+        assert(copy_constructed.has_value());
+        assert(*copy_constructed == 42);
+
+        optional<T> move_constructed{T{42}};
+        assert(move_constructed.has_value());
+        assert(*move_constructed == 42);
+
+        optional<T> copy_assigned;
+        assert(!copy_assigned.has_value());
+        copy_assigned = input;
+        assert(copy_assigned.has_value());
+        assert(*copy_assigned == 42);
+
+        optional<T> move_assigned;
+        assert(!move_assigned.has_value());
+        move_assigned = T{42};
+        assert(move_assigned.has_value());
+        assert(*move_assigned == 42);
+    }
+
+    { // construction from convertible type
+        const int input{42};
+        optional<T> copy_constructed{input};
+        assert(copy_constructed.has_value());
+        assert(*copy_constructed == 42);
+
+        optional<T> move_constructed{42};
+        assert(move_constructed.has_value());
+        assert(*move_constructed == 42);
+
+        optional<T> copy_assigned;
+        assert(!copy_assigned.has_value());
+        copy_assigned = input;
+        assert(copy_assigned.has_value());
+        assert(*copy_assigned == 42);
+
+        optional<T> move_assigned;
+        assert(!move_assigned.has_value());
+        move_assigned = 42;
+        assert(move_assigned.has_value());
+        assert(*move_assigned == 42);
+    }
+
+    { // construction from optional with same type
+        optional<T> constructed{42};
+        assert(constructed.has_value());
+        assert(*constructed == 42);
+
+        optional<T> copy_constructed{constructed};
+        assert(copy_constructed.has_value());
+        assert(*copy_constructed == 42);
+
+        optional<T> move_constructed{move(constructed)};
+        assert(move_constructed.has_value());
+        assert(*move_constructed == 42);
+
+#if !(defined(__clang__) || defined(__EDG__)) // TRANSITION, DevCom-1331017
+        if constexpr (!is_same_v<T, With_nontrivial_destructor>) {
+            if (!is_constant_evaluated()) {
+#endif // TRANSITION
+                optional<T> copy_assigned;
+                assert(!copy_assigned.has_value());
+                copy_assigned = constructed;
+                assert(copy_assigned.has_value());
+                assert(*copy_assigned == 42);
+
+                optional<T> move_assigned;
+                assert(!move_assigned.has_value());
+                move_assigned = move(constructed);
+                assert(move_assigned.has_value());
+                assert(*move_assigned == 42);
+#if !(defined(__clang__) || defined(__EDG__)) // TRANSITION, DevCom-1331017
+            }
+        }
+#endif // TRANSITION, DevCom-1331017
+    }
+
+    { // construction from optional with convertible types

```

Comment:
This appears to be testing only assignment, not construction, from different `optional`s.

Reply:
I guess the comments all say "construction" even when they test assignment, so the issue is that the assignment tests are missing. I can add them.